libc-test/build.rs

```diff
@@ -2922,23 +2896,34 @@ fn test_emscripten(target: &str) {
             // FIXME: emscripten uses different constants to constructs these
             n if n.contains("__SIZEOF_PTHREAD") => true,
 
-            // FIXME: `SYS_gettid` was removed in
-            // emscripten-core/emscripten@6d6474e
+            // No epoll support
+            // https://github.com/emscripten-core/emscripten/issues/5033
+            n if n.starts_with("EPOLL") => true,
+
+            // No ptrace.h
+            // https://github.com/emscripten-core/emscripten/pull/17704
+            n if n.starts_with("PTRACE_") => true,
+
+            // No quota.h
+            // https://github.com/emscripten-core/emscripten/pull/17704
+            n if n.starts_with("QIF_") => true,
+            "USRQUOTA" | "GRPQUOTA" | "Q_GETFMT" | "Q_GETINFO" | "Q_SETINFO" | "Q_SYNC"
+            | "Q_QUOTAON" | "Q_QUOTAOFF" | "Q_GETQUOTA" | "Q_SETQUOTA" => true,
+
+            // `SYS_gettid` was removed in Emscripten v1.39.9
+            // https://github.com/emscripten-core/emscripten/pull/10439
             "SYS_gettid" => true,
 
-            // FIXME: These values have been changed
-            | "POSIX_MADV_DONTNEED" // to 4
-            | "RLIMIT_NLIMITS" // to 16
-            | "RLIM_NLIMITS" // to 16
-            | "IPPROTO_MAX" // to 263
-            | "F_GETLK" // to 5
-            | "F_SETLK" // to 6
-            | "F_SETLKW" // to 7
-            | "O_TMPFILE" // to 65
-            | "SIG_IGN" // -1
-                => true,
+            // No personality.h
+            // https://github.com/emscripten-core/emscripten/pull/17704
+            "ADDR_NO_RANDOMIZE" | "MMAP_PAGE_ZERO" | "ADDR_COMPAT_LAYOUT" | "READ_IMPLIES_EXEC"
+            | "ADDR_LIMIT_32BIT" | "SHORT_INODE" | "WHOLE_SECONDS" | "STICKY_TIMEOUTS"
+            | "ADDR_LIMIT_3GB" => true,
 
-            // LFS64 types have been removed in Emscripten 3.1.44+
+            // FIXME: https://github.com/emscripten-core/emscripten/pull/14883
+            "SIG_IGN" => true,
```

Comment:
What does this FIXME mean? Did that PR break this test? Why?

Reply:
That PR did break the libc test, see e.g.:
```
RUNNING ALL TESTS
bad SIG_IGN value at byte 0: rust: 1 (0x1) != c 254 (0xfe)
bad SIG_IGN value at byte 1: rust: 0 (0x0) != c 255 (0xff)
bad SIG_IGN value at byte 2: rust: 0 (0x0) != c 255 (0xff)
bad SIG_IGN value at byte 3: rust: 0 (0x0) != c 255 (0xff)
thread 'main' panicked at /checkout/target/wasm32-unknown-emscripten/debug/build/libc-test-890b1e2d4768b06a/out/main.rs:12:21:
some tests failed
```

However, this `FIXME` can probably not fixed according to 
https://github.com/emscripten-core/emscripten/pull/14883/files#r892912370.

I'll change this to:
```suggestion
            // `SIG_IGN` has been changed to -2 since 1 is a valid function address
            // https://github.com/emscripten-core/emscripten/pull/14883
            "SIG_IGN" => true,
```

Note that other implementations have a similar issue, see e.g:
https://github.com/rust-lang/libc/blob/e4c7accc68b1a376eb1f538d7b991802793afe96/libc-test/build.rs#L1198

Reply:
Maybe there should be a constant somewhere that indicates what the value of SIG_IGN is on the platform... But that would presumably be dealt with best in a followup.